src/metabase/util/ssh.clj

```diff
@@ -126,16 +132,27 @@
           [session ^PortForwardingTracker tracker] (start-ssh-tunnel! (assoc details :host host))
           tunnel-entrance-port                     (.. tracker getBoundAddress getPort)
           tunnel-entrance-host                     (.. tracker getBoundAddress getHostName)
+          orig-port                                (:port details)
           details-with-tunnel                      (assoc details
                                                           :port tunnel-entrance-port ;; This parameter is set dynamically when the connection is established
                                                           :host (str proto "localhost") ;; SSH tunnel will always be through localhost
+                                                          :orig-port orig-port
                                                           :tunnel-entrance-host tunnel-entrance-host
                                                           :tunnel-entrance-port tunnel-entrance-port ;; the input port is not known until the connection is opened
+                                                          :tunnel-enabled true
                                                           :tunnel-session session
                                                           :tunnel-tracker tracker)]
       details-with-tunnel)
     details))
 
+(defmethod driver/incorporate-ssh-tunnel-details nil
```

Comment:
why would a driver ever be nil? I don't think this could or should happen

Reply:
Some test cases for databases (either in `api` or `models`) had such a situation.  Rather than trying to go down the rabbit hole of rewriting them (I think they were using some contrived test engines that somehow didn't fully register), this seemed to be the easier fix.

Reply:
Actually, seems unnecessary (removed).